docs/course_materials/exercises/test_exercises/04_Compound_data_types/test_homework_dictionary.py

```diff
@@ -0,0 +1,72 @@
+"""
+Automatically check student's homework
+----------------------------------------
+The tests will check:
+ 1. The content of the exercise file
+ 2. The execution and output of the exercise file
+ 3. The output with different inputs
+
+Guidelines:
+- You have to specify the dir Student's exercise files should be placed
+in the same dir as this test.
+- Test names are in the form: test_<exercise>.py
+- Before each test should be the appropriate fixture: @pytest.mark.<topic>
+"""
+import pytest
+
+
+@pytest.mark.dictionary
+def test_countries(helpers):
+    helpers.set_student_file('countries.py')
+    helpers.expected_pycode = [
+        # Create a new dictionary with Udi’s flights number to each country:
+        # "Italy":2 "Spain":3 "Israel":1
+        [r'''^.*\s*=\s*\{.*\bItaly\b.*\:\s?2\s?,.*\bSpain\b.*\:\s?3\s?,''' +
+         r'''.*\bIsrael\b.*\:\s?1\s?\}''',
+         'set the dictionary using: {"Italy":2 "Spain":3 "Israel":1}'],
+        # Insert a new element : "Belgium":1
+        [r'''^.*\[(('Belgium')|("Belgium"))\]\s*=\s*1''',
+         'to add a key-value pair use: `dict["key"] = value`'],
+        # Delete “Italy” record from the dictionary
+        [r'''^del\s*.*\[(('Italy')|("Italy"))\]$''',
+         'to delete a key-value pair use: `del dict["key"]'],
+        # Print how many times Udi was in Spain
+        [r'''print\(.*\[(('Spain')|("Spain"))\]\)''',
+         'to print a value use: `print(dict["key"])`'],
+    ]
+
+    helpers.test_assignment()
+
+
+@pytest.mark.dictionary
+def test_family(helpers):
+    helpers.set_student_file('family_members_average_age.py')
+    helpers.expected_pycode = [
+        # Define a dictionary called ages, ages should be empty in it this time
+        [r'''^ages\s*=\s*\{\}$''',
+         'define a dictionary using `ages = {}`'],
+        # Add your family members names to the dictionary with their ages.
+        # for example: “Samira”:8
+        [r'''(ages\.update\({)|(ages\[.*\]=\d*)''',
+         'add a member using dict.update() or dict(key)=value'],
```

Comment:
also a good student work that failes:
ages = {}
ages = {'Ran' : 50, 'Adi' : 49, 'Oren' : 22, 'Shira' : 20, 'Erez' : 17}
print (ages)
print((ages['Oren'] + ages['Shira'] + ages['Erez'] + ages['Ran'] + ages['Adi'])/5)

failures: 
-------------------------------------------------- live log call --------------------------------------------------
[    INFO] Started testing /home/mshira/Downloads/Erez/family_members_average_age.py:
[    INFO] testing the code...
[ WARNING] 
add a member using dict.update() or dict(key)=value
[    INFO] testing the output...
FAILED                                                                                                      [100%]

also- is it possible to check here if the student use len(ages) for calculating the average?

Reply:
Setting the dictionary - this solution works, but wasn't the point to use update or assigning, maybe we should add a hint in the exercise? Also, if I understand correctly, the command used just rewrites the variable.
Len - the regex is set to allow both: len() and a number.
Please let me know which option you prefer for those cases.

Reply:
fixed